### PR TITLE
sharkychatbot

### PR DESCRIPTION
**Pull request by vittoriamatteoli**
## Netlify link
[Add your Netlify link here.
PS. Don't forget to add it in your readme as well.](https://sharkychatbot.netlify.app/)

## Collaborators
Add your collaborators here. Write their GitHub usernames in square brackets. If there's more than one, separate them with a comma, like this:
[vittoriamatteoli, jingh999]

### CODE REVIEW

File: code/script.js [hunk 1]
```diff
@@ -1,53 +1,230 @@
...
+// Moved the message function here 👇🏻
 
-// A function that will add a chat bubble in the correct place based on who the sender is
 const showMessage = (message, sender) => {
-  // The if statement checks if the sender is the user and if that's the case it inserts
-  // an HTML section inside the chat with the posted message from the user
-  if (sender === 'user') {
+  if (sender === "user") {
+    console.log("message is:", message)
+    console.log("sender is:", sender)
```
Comment (HIPPIEKICK) [inline code/script.js]: Remove console.logs, they are used for development and shouldn't be in production versions.

File: code/script.js [hunk 2]
```diff
@@ -1,53 +1,230 @@
...
-  // Just to check it out, change 'bot' to 'user' here 👆 and see what happens
+const handleNameInput = (event) => {
+  event.preventDefault()
+
+  // Store the value in a variable so we can access it after we
+  // clear it from the input
+
+  const nameInput = document.getElementById("name-input")
+  const name = nameInput.value
+  if (/^[a-zA-z]/.test(name)) {
```
Comment (HIPPIEKICK) [inline code/script.js]: Nice that you want to try out regex, but it doesn't seem to work. It accepts digits and empty inputs right now

File: code/script.js [hunk 3]
```diff
@@ -1,53 +1,230 @@
...
+    setTimeout(() => firstChoice(name), 500)
+  } else {
+    showMessage(`Please fill in a valid name with letters.`, "bot")
+  }
 }
 
-// Eventlisteners goes here 👇
+// second message from bot
+
+const firstChoice = (name) => {
```
Comment (HIPPIEKICK) [inline code/script.js]: Maybe this function could get a more descriptive name, like askForHairCutLength?

File: code/script.js [hunk 4]
```diff
@@ -1,53 +1,230 @@
...
+  you like to have ?`,
+    `bot`
+  )
+
+  // added the 3 buttons for choice
+
+  inputWrapper.innerHTML = `<div id="button-form">
+  <button id="short">Short</button>
+  <button id="medium">Medium</button>
+  <button id="long">Long</button></div>`
```
Comment (HIPPIEKICK) [inline code/script.js]: Make sure to always format your HTML properly, even if the HTML is in the JS file:
```
<div id="button-form">
  <button id="short">Short</button>
  <button id="medium">Medium</button>
  <button id="long">Long</button>
</div>
```

File: code/script.js [hunk 5]
```diff
@@ -1,53 +1,230 @@
...
+    showMessage(`Sorry to see you go. You can always book again!`, `bot`)
+    setTimeout(resetConversation, 1500)
+  }
+}
+// Fixed to reload the page
+const resetConversation = () => {
+  chat.innerHTML = ""
+  inputWrapper.innerHTML = ""
+  setTimeout(greetUser, 2000)
+}
```
Comment (HIPPIEKICK) [inline code/script.js]: Nice!

**Review by HIPPIEKICK — APPROVED** [review]
Good job with the chatbot! I made some small comments to think about, but overall I think your code is clean and structured and your bot works well.

File: .vscode/settings.json
Comment (NellaMori) [inline .vscode/settings.json:1]: Delete this folder unless you want to share the settings with others, you can safely add it to your gitignore file for example 🧹

File: .DS_Store
Comment (NellaMori) [inline .DS_Store:1]: .DS_Store is a file that stores custom attributes of its containing folder, such as folder view options, icon positions, and other visual information, and we don't need to share it in our project. Please add .DS_Store to your .gitignore file (if you don't have one, you need to create it) 👍

File: code/.DS_Store
Comment (NellaMori) [inline code/.DS_Store:1]: delete this file here too 👍

File: code/index.html [hunk 6]
```diff
@@ -1,32 +1,42 @@
...
-            Send
-          </button>
-        </form>
-      </div>
+      <div
+        class="input-wrapper button-wrapper select-wrapper"
+        id="input-wrapper"
+      ></div>
+      <!-- audio here 👇🏻 -->
+      <audio id="botAudio" typ="audio/mpeg" src="./assets/pop.mp3"></audio>
```
Comment (NellaMori) [inline code/index.html:33]: such a nice addition with the sound effect! 😍

File: code/script2.js [hunk 7]
```diff
@@ -0,0 +1,276 @@
...
+  const nameInput = document.getElementById("name-input")
+  const name = nameInput.value
+  showMessage(`I'm ${name}!`, "user")
+  nameInput.value = ""
+  setTimeout(() => askNextQuestion(name), 500)
+}
+
+// second message from bot
+
+const AskTypeHaircut = (name, choice) => {
```
Comment (NellaMori) [inline code/script2.js]: this name should follow camelCase pattern 👍

**Review by AntonellaMorittu — APPROVED** [review]
Hi Vittoria & Jing good job this week, you can tell you had fun working on this project! The code meets the requirements, what you can add to continue improving the UX is to add name field validation, you add it there but it's not working just yet. We left some comments specifically on some parts, please take a look at them :) Thank you ⭐

**Pull request closed without merging**